docs/concepts/crds/restoresession.md

```diff
@@ -0,0 +1,248 @@
+    parent: crds
+    weight: 25
+product_name: stash
+menu_name: product_stash_0.8.3
+section_menu_id: concepts
+---
+
+> New to Stash? Please start [here](/docs/concepts/README.md).
+
+# RestoreSession
+
+## What is RestoreSession
+
+A `RestoreSession` is a Kubernetes `CustomResourceDefinition`(CRD) which specifies where to restore (target) and the source of data that will be restored in a Kubernetes native way.
+
+You have to create a `RestoreSession` object whenever you want to restore. When a `RestoreSession` object is created, Stash injects an `init-container` into the target workload and restart it. The `init-container` restore the desired data. If the target is a database or a stand-alone PVC, Stash lunch a job to perform the restore process.
+
+>In this tutorial, we are going to use **host** word to indicate an entity (pod) where data is restored.
+
+>- For `Deployment`, `ReplicationController` and `ReplicaSet`, restore process run in only one pod. This pod is referred as **host-0**.
+>- For `StatefulSet`, restore process runs in all pods. In this case, **pod-0** is known as **host-0**, **pod-1** is known as **host-1**, **pod-2** is known as **host-2** and so on.
+>- For `DaemonSet`, restore process runs in all daemon pods. In this case, the **node name** where the pod is running act as their **host** name.
+>- For database or stand-alone PVC, restore is done by a job. In this case, the restore job's pod is known as **host-0**.
+
+## RestoreSession CRD Specification
+
+Like any official Kubernetes resource, a `RestoreSession` has `TypeMeta`, `ObjectMeta`, `Spec` and `Status` sections.
+
+A sample `RestoreSession` object to restore backed up data of a StatefulSet is shown below,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: RestoreSession
+metadata:
+  name: statefulset-restore
+  namespace: demo
+spec:
+  driver: Restic
+  repository:
+    name: local-repo
+  # task:
+  #   name: workload-restore # task field is not required for workload data restore but it is necessary for database restore.
+  target:
+    ref:
+      apiVersion: apps/v1
+      kind: StatefulSet
+      name: recovered-statefulset
+    volumeMounts:
+    - mountPath: /source/data
+      name: source-data
+  rules:
+  - subjects: ["host-3","host-4"] # "host-3" and "host-4" will have restored data of backed up host "host-1"
+    sourceHost: "host-1" # source host
+    paths:
+    - /source/data
+  - subjects: [] # empty host match all hosts
+    sourceHost: "" # no source host indicates that the host is pod itself
+    paths:
+    - /source/data
+  runtimeSettings:
+    container:
+      resources:
+        limits:
+          memory: 256M
+        requests:
+          memory: 256M
+      securityContext:
+        runAsGroup: 2000
+        runAsUser: 2000
+      ionice:
+        class: 2
+        classData: 4
+      nice:
+        adjustment: 5
+    pod:
+      imagePullSecrets:
+      - name: my-private-registry-secret
+      serviceAccountName: my-restore-svc
```

Comment:
Do we give validation error when serviceAccountName is used for worklooad target ?

Reply:
invalid example.